featuretools/tests/entityset_tests/test_es.py

```diff
@@ -36,80 +78,173 @@ def test_cannot_re_add_relationships_that_already_exists(es):
-        'value': variable_types.Numeric,
-        'value_2': variable_types.Numeric,
-        'latlong': variable_types.LatLong,
-        'latlong2': variable_types.LatLong,
-        'zipcode': variable_types.ZIPCode,
-        'countrycode': variable_types.CountryCode,
-        'subregioncode': variable_types.SubRegionCode,
-        'value_many_nans': variable_types.Numeric,
-        'priority_level': variable_types.Ordinal,
-        'purchased': variable_types.Boolean,
-        'comments': variable_types.NaturalLanguage
+        parent_df = es[r.parent_dataframe.ww.name]
+        child_df = es[r.child_dataframe.ww.name]
+        assert parent_df.ww.index == r.parent_column.name
+        assert 'foreign_key' in r.child_column.ww.semantic_tags
+        assert str(parent_df[r.parent_column.name].dtype) == str(child_df[r.child_column.name].dtype)
+
+
+def test_add_relationship_instantiated_logical_types(es):
+    log_2_df = es['log'].copy()
+    log_logical_types = {
+        'id': ltypes.Integer,
+        'session_id': ltypes.Integer,
+        'product_id': ltypes.Categorical(),
+        'datetime': ltypes.Datetime,
+        'value': ltypes.Double,
+        'value_2': ltypes.Double,
+        'latlong': ltypes.LatLong,
+        'latlong2': ltypes.LatLong,
+        'zipcode': ltypes.PostalCode,
+        'countrycode': ltypes.CountryCode,
+        'subregioncode': ltypes.SubRegionCode,
+        'value_many_nans': ltypes.Double,
+        'priority_level': ltypes.Ordinal(order=[0, 1, 2]),
+        'purchased': ltypes.Boolean,
+        'comments': ltypes.NaturalLanguage
     }
-    assert set(log_variable_types) == set(log_2_df.columns)
-    es.entity_from_dataframe(entity_id='log2',
-                             dataframe=log_2_df,
-                             index='id',
-                             variable_types=log_variable_types,
-                             time_index='datetime')
-
-    error_text = u'Unable to add relationship because id in customers is Pandas dtype category and session_id in log2 is Pandas dtype int64.'
-    with pytest.raises(ValueError, match=error_text):
+    log_semantic_tags = {
+        'session_id': 'foreign_key',
+        'product_id': 'foreign_key'
+    }
+    assert set(log_logical_types) == set(log_2_df.columns)
+    es.add_dataframe(dataframe_name='log2',
+                     dataframe=log_2_df,
+                     index='id',
+                     logical_types=log_logical_types,
+                     semantic_tags=log_semantic_tags,
+                     time_index='datetime')
+    assert 'log2' in es.dataframe_dict
+    assert es['log2'].ww.schema is not None
+    assert es['log2'].ww.logical_types['product_id'] == ltypes.Categorical()
+    assert es['products'].ww.logical_types['id'] == ltypes.Categorical
+
+    warning_text = 'Logical type Categorical for child column product_id does not match parent '\
+        'column id logical type Categorical. Changing child logical type to match parent.'
```

Comment:
The difference here is that one logical type is instantiated and the other is not. There's a Woodwork change in progress that's actually going to have all logical types instantiated, so this test will probably stop raising an error once that's in main.

But your point still stands because if the difference was parameters, we'd get the same error that'd be confusing. I'll update the error message to include the logical type's parameters if there are any.

Reply:
Added an element to the warning that will indicate if the parameters do not match.